reana_client/cli/workflow.py

```diff
@@ -1458,3 +1464,84 @@ def workflow_close_interactive_session(workflow, access_token):  # noqa: D301
             sys.exit(1)
     else:
         display_message("Cannot find workflow {} ".format(workflow), msg_type="error")
+
+
+@workflow_sharing_group.command("share-add")
+@check_connection
+@add_workflow_option
+@add_access_token_options
+@click.option(
+    "-u",
+    "--user",
+    "users",
+    multiple=True,
+    help="Users to share the workflow with.",
+    required=True,
+)
+@click.option(
+    "-m",
+    "--message",
+    help="Optional message that is sent to the user(s) with the sharing invitation.",
+)
+@click.option(
+    "-v",
+    "--valid-until",
+    help="Optional date when access to the workflow will expire for the given user(s) (format: YYYY-MM-DD).",
+)
+@click.pass_context
+def share_workflow_add(
+    ctx, workflow, access_token, users, message, valid_until
+):  # noqa D412
+    """Share a workflow with other users (read-only).
+
+    The `share-add` command allows sharing a workflow with other users. The
+    users will be able to view the workflow but not modify it.
+
+    Examples:
+
+    $ reana-client share-add -w myanalysis.42 --user bob@example.org
+
+    $ reana-client share-add -w myanalysis.42 --user bob@example.org --user cecile@example.org --message "Please review my analysis" --valid-until 2025-12-31
+    """
+    from reana_client.api.client import share_workflow
+
+    share_errors = []
+    shared_users = []
+
+    if workflow:
```

Comment:
No need to do `if workflow: ... else: display_message(...)`, as this is already handled by `@add_workflow_option`. I think there are still some commands that do this, but it's a leftover from the past (that we should probably clean up at some point :sweat_smile: )

Reply:
Nicely spotted. Changed

The changes were applied to the following superseding PR: #692